lookout/style/typos/analyzer.py

```diff
@@ -107,18 +135,31 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
                 for index in suggestions.keys():
                     corrections = suggestions[index]
                     for token in corrections.keys():
-                        comment = Comment()
-                        comment.file = file.path
-                        corrections_line = " " + ", ".join(
-                            "%s (%d%%)" % (candidate[0], int(candidate[1] * 100))
-                            for candidate in corrections[token])
-                        comment.text = """
-                            Possible typo in \"%s\". Suggestions:
-                        """.strip() % new_identifiers[index].token + corrections_line
-                        comment.line = new_identifiers[index].start_position.line
-                        comment.confidence = int(corrections[token][0][1] * 100)
-                        comments.append(comment)
-        return comments
+                        yield TypoFix(
+                            head_file=file,
+                            token=new_identifiers[index].token,
+                            candidates=[Candidate(*c[:2]) for c in corrections[token]],
+                            line_number=new_identifiers[index].start_position.line,
+                        )
+
+    def render_comment_text(self, typo_fix: TypoFix) -> str:
+        """
+        Generate the text of the comment for the specified typo fix.
+
+        :param typo_fix: Information about typo fix required to render a comment text.
+        :return: string with the generated comment.
+        """
+        # TODO: move this logic to template. See FormatAnalyzer.render_comment_text()
+        corrections_line = ", ".join("%s (%d%%)" % (
+            c.token, int(c.confidence * 100)) for c in typo_fix.candidates)
+        return "Possible typo in \"%s\". Suggestions: %s" % (typo_fix.token, corrections_line)
+
+    @staticmethod
+    def _get_comment_confidence(suggestions: Sequence[Candidate]) -> int:
+        invert_confidence = 1
+        for suggestion in suggestions:
+            invert_confidence *= (1 - suggestion.confidence)
```

Comment:
Looks like here you multiply probabilities for correcting the same token inside on identifier, while wanted to do it for the whole identifier.
So for example you have `get nomber of noddes`. Model's corrections for this identifier are `nomber: (number, 0.99), (none, 0.10)` and `noddes: (nodes, 0.9), (node, 0.6)`.
And here you return comments `Typo in 'nomber' with confidence (1 - 0.01 * 0.9)` and `Typo in 'noddes' with confidence (1 - 0.1 * 0.4)`. 
Is that exactly what you wanted to do, not multiply probas for different corrections for the whole identifier? Like `Typo in 'get_nomber_of_noddes with confidence (1 - 0.01 * 0.1)`?
If you did want the first way, then why not to take the best candidate probability as the confidence then?

Reply:
you are right, it is a probability of having a good suggestion for one token in the identifier. And for now, we have two comments for your example. One per typo. 

So actually, I keep it as it was and we should change it is the future when we are able to reconstruct identifier. Thus, in next PRs. :)

Reply:
another point - probabilities are not normalized `noddes: (nodes, 0.9), (node, 0.6)`.

Reply:
well, we will change it to anything we want when it becomes important. For now, it is ok to have any reasonable number. Anyway we cannot verify it now.

Reply:
it is fine they are not normalized. That is why I use black probability theory magic and not just sum it up.